File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzseek.c
z_off_t ZEXPORT gzseek(gzFile file, z_off_t offset, int whence) {
    z_off64_t ret;

    ret = gzseek64(file, (z_off64_t)offset, whence);
    return ret == (z_off_t)ret ? (z_off_t)ret : -1;
}

File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzseek.rs
rust
extern crate libc;

use libc::{c_long, c_ulong};

fn gzseek(file: *mut u8, offset: i64, whence: i32) -> i64 {
    let ret: i64 = unsafe { libc::gzseek64(file, offset as u64, whence as u32) };
    if ret == offset as u64 {
        return ret;
    } else {
        return -1;
    }
}